### DISCUSSION TITLE
How to hide the download button when download is not available

### DISCUSSION THREAD

**Discussion by ash67** [igordanchenko/yet-another-react-lightbox#260] [Q&A] [answered]
I would like to hide the download button for the slides which are not non-downloadable. When setting `dowload: false`, it does show the disabled icon which is cool, but I want to completely hide it. Is that possible?

Also, when rendering customIcon, is it possible to use a different icon when hover/clicking on the download button. 

Any help would be greatly appreciated.  Thanks!

**Comment by igordanchenko** [comment, marked as answer]
> I would like to hide the download button for the slides which are not non-downloadable. When setting dowload: false, it does show the disabled icon which is cool, but I want to completely hide it. Is that possible?

Sure, you can hide the inactive Download button with the following CSS customization:

```css
.yarl__button[aria-label="Download"]:disabled {
  display: none;
}
```

> Also, when rendering customIcon, is it possible to use a different icon when hover/clicking on the download button.

Not unless you implement your own [DownloadButton](https://github.com/igordanchenko/yet-another-react-lightbox/blob/main/src/plugins/download/DownloadButton.tsx) and render it through `render.buttonDownload`. 

However, you can target `:hover` and `:active` states and provide certain customizations with CSS.

```css
.yarl__button[aria-label="Download"]:hover {
  /* ... */
}

.yarl__button[aria-label="Download"]:active {
  /* ... */
}
```

**Reply by ash67** [reply]
Thanks so much!
